nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/bigquery/PutBigQueryBatch.java

```diff
@@ -0,0 +1,269 @@
+            .displayName("Ignore Unknown Values")
+            .description(BigQueryAttributes.IGNORE_UNKNOWN_DESC)
+            .required(true)
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .allowableValues("true", "false")
+            .defaultValue("true")
+            .build();
+
+    public static final PropertyDescriptor CREATE_DISPOSITION = new PropertyDescriptor.Builder()
+            .name(BigQueryAttributes.CREATE_DISPOSITION_ATTR)
+            .displayName("Create Disposition")
+            .description(BigQueryAttributes.CREATE_DISPOSITION_DESC)
+            .required(true)
+            .allowableValues(JobInfo.CreateDisposition.CREATE_IF_NEEDED.name(), JobInfo.CreateDisposition.CREATE_NEVER.name())
+            .defaultValue(JobInfo.CreateDisposition.CREATE_IF_NEEDED.name())
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor WRITE_DISPOSITION = new PropertyDescriptor.Builder()
+            .name(BigQueryAttributes.WRITE_DISPOSITION_ATTR)
+            .displayName("Write Disposition")
+            .description(BigQueryAttributes.WRITE_DISPOSITION_DESC)
+            .required(true)
+            .allowableValues(JobInfo.WriteDisposition.WRITE_EMPTY.name(), JobInfo.WriteDisposition.WRITE_APPEND.name(), JobInfo.WriteDisposition.WRITE_TRUNCATE.name())
+            .defaultValue(JobInfo.WriteDisposition.WRITE_EMPTY.name())
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor MAXBAD_RECORDS = new PropertyDescriptor.Builder()
+            .name(BigQueryAttributes.MAX_BADRECORDS_ATTR)
+            .displayName("Max Bad Records")
+            .description(BigQueryAttributes.MAX_BADRECORDS_DESC)
+            .required(true)
+            .defaultValue("0")
+            .addValidator(StandardValidators.NON_NEGATIVE_INTEGER_VALIDATOR)
+            .build();
+
+    private Schema schemaCache = null;
+
+    public PutBigQueryBatch() {
+
+    }
+
+    @Override
+    public List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return ImmutableList.<PropertyDescriptor>builder()
+                .addAll(super.getSupportedPropertyDescriptors())
+                .add(DATASET)
+                .add(TABLE_NAME)
+                .add(TABLE_SCHEMA)
+                .add(SOURCE_TYPE)
+                .add(CREATE_DISPOSITION)
+                .add(WRITE_DISPOSITION)
+                .add(MAXBAD_RECORDS)
+                .add(IGNORE_UNKNOWN)
+                .build();
+    }
+
+    @Override
+    protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String propertyDescriptorName) {
+        return new PropertyDescriptor.Builder()
+                .name(propertyDescriptorName)
+                .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+                .expressionLanguageSupported(true)
```

Comment:
.expressionLanguageSupported(boolean) is deprecated. Can you please specify the scope if dynamic properties are actually used somewhere?